### DISCUSSION TITLE
New borg install : adding lsp mode - clients are not accessible

### DISCUSSION THREAD

**Discussion by rileyrg** [emacscollective/borg#105] [Support] [answered]
I just set up my first borg config : I didnt use the seed bootstrap but rather the from scratch method

https://github.com/emacscollective/borg/blob/master/borg.org#bootstrapping-from-scratch

You can see my init.el [here](https://github.com/rileyrg/Emacs-Customisations/blob/borg-full/init.el) 

I went through my config and run borg-assimilate for each use-package and then spent another while doing the same for packages it said was missing when I started emacs up.

All went reasonably well.

But, I'm unable to require any clients from the lib/lsp-mode/clients subdirectory. This results in not being able to install any lsp servers. 

`lsp-auto-configure` is t.

eg (require 'lsp-clangd)

I know I could add this to the path but I would rather not. Is there something I have forgotten (I ran make)? Or is this a problem with lsp autoloads or some such?

**Comment by tarsius** [comment, marked as answer]
You'll have to set the load-path explicitly:

```
[submodule "lsp-mode"]
	path = lib/lsp-mode
	url = git@github.com:emacs-lsp/lsp-mode.git
	load-path = .
	load-path = clients
```

**Reply by eki3z** [reply]
Can `load-path = .` be added by default ? 
Every time I add extra directories for a package , I need to add that first. That make gitmodules looks redundant. I guess every package creator would agree with it.

**Reply by tarsius** [reply]
No that's not possible because for some packages it would be wrong to add `load-path = .` in addition to `load-path = whatever`. By making it necessary to add explicitly add `.` it's possible to handle all three cases.

**Reply by eki3z** [reply]
Or could the format be designed like `load-path = (.,clients,extensions/elisp)`? I would prefer a one-line style, use comma as seperate

**Reply by dakra** [reply]
I'm not for or against a different load-path syntax but just as a reference:
I have almost 400 packages in [my config](https://github.com/dakra/dmacs/blob/master/.gitmodules) and only had to type 3 times `load-path = .`.
So I don't think it's a big or common problem.

**Reply by tarsius** [reply]
@liuyinz No, I prefer to stick to the form intended by the authors of this file format (the git authors). I also won't support two different forms because I want to keep the "convention over configuration" spirit of this package alive. We gain a lot by not implementing features until we are certain there is a real need.

**Reply by eki3z** [reply]
I look up mine: 150 package with 2 times: vertico and lsp-mode...yep, it's not a big problem.

**Reply by eki3z** [reply]
@tarsius understood.
